### PR TITLE
crop: fix issue with handle confusion when going from rotation->cropping

### PR DESCRIPTION
**Pull request by mimecuvalo**
fixes https://github.com/tldraw/tldraw/issues/4845
we shouldn't keep rotate handles around as it's confusing UI - it's not really what's going to happen when the ctrlKey is being pressed.

https://github.com/user-attachments/assets/74f6063d-9ef0-499b-8222-856b51d86f7a

### Change type

- [x] `bugfix`
- [ ] `improvement`
- [ ] `feature`
- [ ] `api`
- [ ] `other`

### Release notes

- Update the handles when we're going to enter into a cropping mode.

### CODE REVIEW

File: packages/tldraw/src/lib/canvas/TldrawSelectionForeground.tsx [hunk 1]
```diff
@@ -35,6 +38,22 @@ export const TldrawSelectionForeground = track(function TldrawSelectionForegroun
...
+			}
+		}
+		editor.addListener('event', handleEvent)
+		return () => {
+			editor.removeListener('event', handleEvent)
+		}
+	}, [editor, isCtrlKeyPressed])
+
+	const canEnterCropMode = isCtrlKeyPressed && getShouldEnterCropMode(editor)
+
```
Comment (steveruizok) [inline packages/tldraw/src/lib/canvas/TldrawSelectionForeground.tsx:56]: We should really make editor.inputs into an atom, huh

Comment (steveruizok) [reply packages/tldraw/src/lib/canvas/TldrawSelectionForeground.tsx:56]: Just took a shot at it, that would be a big old change, let's land this and then think about that.

**Review by steveruizok — DISMISSED** [review]

**Comment by mimecuvalo** [comment]
> Nevermind, I don't like this. The crop UI shows up too often, e.g. when pressing copy and paste. We show the handles when control+clicking the edges and after double clicking the shape, I think that's enough.

We still need to address the bug that this is causing though: https://github.com/tldraw/tldraw/issues/4845
(that was the original impetus for this PR)

**Comment by steveruizok** [comment]
Is that fix in this PR? or should we treat it as a fresh problem?

ps for vibe / attitude etc, this PR works and is a good fix. When user intents overlap like this (here, is the cmd key down because the user is pressing a modifier for a shortcut, or are they trying to crop an image?) I try not to change anything in the UI as it'll be "noticeably wrong" in cases where the user has intent A instead of intent B, even if it hurts the discoverability of intent B.

**Comment by mimecuvalo** [comment]
> Is that fix in this PR? or should we treat it as a fresh problem?

Yeah, the fix is in this PR — it's that you don't go accidentally into the rotate mode which causes a crash when holding `Ctrl`

> ps for vibe / attitude etc, this PR works and is a good fix. I have some poorly formed design principles around "collisions" baked into tldraw. When user intents overlap like this (here, is the cmd key down because the user is pressing a modifier for a shortcut, or are they trying to crop an image?) I try not to change anything in the UI as it'll be "noticeably wrong" in cases where the user has intent A instead of intent B, even if it hurts the discoverability of intent B.

Cool, no worries, I hear that. I think my take on it is similar, say, to when you're dragging an arrow and toggling `Cmd` (toggling binding) - the UI does change from underneath you to show intent. 

Let's discuss more offline and come up with a solution!

**Comment by steveruizok** [comment]
PR here to fix https://github.com/tldraw/tldraw/pull/5087

**Pull request closed without merging**